File: c_sharp/Enumeration/Enumeration/Training/TrainingTypes.cs
using PeerageHRIS.Resources;
using PeerageHRIS.Resources.Models.Training;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration.Training
{
    public enum TrainingTypes
    {
        [Display(Name = "ShortTerm", ResourceType = typeof(TrainingResource))]
        ShortTerm,
        [Display(Name = "MidTerm", ResourceType = typeof(TrainingResource))]
        MidTerm,
        [Display(Name = "LongTerm", ResourceType = typeof(TrainingResource))]
        LongTerm,
        [Display(Name = "Other", ResourceType = typeof(TrainingResource))]
        Other
    }
}

File: lib/core/enums/training_types.dart
import 'package:json_annotation/json_annotation.dart';

part 'training_types.g.dart';

@JsonEnum(alwaysCreate: true)
enum TrainingTypes {
  @JsonValue(0)
  shortTerm,

  @JsonValue(1)
  midTerm,

  @JsonValue(2)
  longTerm,

  @JsonValue(3)
  other,
}
